src/foundry/foundry.js/applications/basePlaceableHUD.d.ts

```diff
@@ -1,93 +1,89 @@
+ * @typeParam Options - the type of the options object
  */
 declare abstract class BasePlaceableHUD<
-  O extends PlaceableObject<any>,
-  P extends Application.Options = Application.Options
-> extends Application<P> {
+  Object extends PlaceableObject<any> = PlaceableObject,
+  Options extends Application.Options = Application.Options
+> extends Application<Options> {
   /**
    * Reference a PlaceableObject this HUD is currently bound to
-   * @defaultValue `null`
+   * @defaultValue `undefined`
    */
-  object: O | null;
+  object: Object | undefined;
 
   /**
    * @override
    * @defaultValue
    * ```
-   * mergeObject(super.defaultOptions, {
+   * foundry.utils.mergeObject(super.defaultOptions, {
    *   classes: ["placeable-hud"],
    *   popOut: false
    * })
    * ```
    */
-  static get defaultOptions(): typeof Application['defaultOptions'];
+  static get defaultOptions(): Application.Options;
 
   /**
    * Convenience access for the canvas layer which this HUD modifies
    */
-  get layer(): O['layer'];
+  get layer(): Object['layer'] | undefined;
 
   /**
    * Bind the HUD to a new PlaceableObject and display it
    * @param object - A PlaceableObject instance to which the HUD should be bound
    */
-  bind(object: O): void;
+  bind(object: Object): void;
 
   /**
    * Clear the HUD by fading out it's active HTML and recording the new display state
    */
   clear(): void;
 
   /** @override */
-  _render(...args: Parameters<Application<P>['_render']>): Promise<void>;
+  _render(...args: Parameters<Application<Options>['_render']>): Promise<void>;
 
-  /**
-   * @override
-   * @defaultValue
-   * ```
-   * mergeObject(this.object.data, {
-   *   id: this.id,
-   *   classes: this.options.classes.join(' '),
-   *   appId: this.appId,
-   *   isGM: game.user.isGM,
-   *   icons: CONFIG.controlIcons
-   * });
-   * ```
-   */
-  getData(options?: Partial<P>): foundry.utils.Duplicated<O['data']> & {
-    id: Application['id'];
+  /** @override */
+  getData(options?: Partial<Options>): ReturnType<Object['data']['toJSON']> & {
+    id: BasePlaceableHUD['id'];
     classes: string;
-    appId: Application['appId'];
+    appId: BasePlaceableHUD['appId'];
     isGM: boolean;
     icons: typeof CONFIG['controlIcons'];
   };
 
-  /**
-   * @override
-   * @remarks Returns `void`
-   */
-  setPosition({ left, top, width, height, scale }?: Partial<Application.Position>): any;
+  /** @override */
+  setPosition({ left, top, width, height, scale }?: Partial<Application.Position>): undefined;
 
   /** @override */
-  activateListeners(html: JQuery<HTMLElement>): void;
+  activateListeners(html: JQuery): void;
+
+  /**
+   * Handle mouse clicks to control a HUD control button
+   * @param event - The originating click event
+   * @remarks This will always return a promise with documents is overridden by TokenHUD.
+   */
+  protected _onClickControl(event: JQuery.ClickEvent): Promise<Array<Object['document']> | void | boolean> | void;
```

Comment:
Maybe just make the return type `unknown` or `void` to signify that it doesn't matter? That would also give deriving classes more freedom.

Basically, I think that for many of the `_on...` functions, we shouldn't really care about the return type as it is not meant to be used. Usually, the return is just there as an early exit from the functions. In particular, usually the return type isn't documented by foundry in these cases.